src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROWTH_CAPACITY = 1.5;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == array.length) {
+            grow();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index "
+                    + index + " is not valid for size " + size + ".");
+        }
+        if (size == array.length) {
+            grow();
+        }
 
+        if (index < size) {
+            System.arraycopy(array, index, array, index + 1, size() - index);
+            array[index] = value;
+            size++;
+        }
+        if (index == size) {
+            add(value);
+        }
```

Comment:
Why bring this decision back? Write the answer in slack, it's not convenient to correspond here